node/subsystem/src/lib.rs

```diff
@@ -210,18 +210,42 @@ pub struct SpawnedSubsystem {
 /// [`SubsystemError`]: struct.SubsystemError.html
 pub type SubsystemResult<T> = Result<T, SubsystemError>;
 
+/// A sender used by subsystems to communicate with other subsystems.
+///
+/// Each clone of this type may add more capacity to the bounded buffer, so clones should
+/// be used sparingly.
+#[async_trait]
+pub trait SubsystemSender: Send + Clone + 'static {
+	/// Send a direct message to some other `Subsystem`, routed based on message type.
+	async fn send_message(&mut self, msg: AllMessages);
+
+	/// Send multiple direct messages to other `Subsystem`s, routed based on message type.
+	async fn send_messages<T>(&mut self, msgs: T)
+		where T: IntoIterator<Item = AllMessages> + Send, T::IntoIter: Send;
+
+	/// Send a message onto the unbounded queue of some other `Subsystem`, routed based on message
+	/// type.
+	///
+	/// This function should be used only when there is some other bounding factor on the messages
+	/// sent with it. Otherwise, it risks a memory leak.
+	fn send_unbounded_message(&mut self, msg: AllMessages);
```

Comment:
feels like it's missing a motivation for this change and a guide how it's supposed to be used, basically turns comments in Elements into the docs

Reply:
We have discussed motivation, no? There's no way to solve the deadlock without having tools for breaking cycles.

Reply:
I meant putting the motivation in the doc/mod comments for future maintainers/new joiners. It's not possible to hold every Element conversation in head over a long period of time.

Reply:
ok, added